### PR TITLE
Selector factory test + d.ts

### PR DESCRIPTION
**Pull request by munkk**

### CODE REVIEW

File: src/core/dom/selector/selector.factory.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {Bag} from '../bag';
+import {SelectorMark} from './selector.mark';
+import {SelectorMediator} from './selector.mediate';
+
+export declare class SelectorFactory {
+	constructor(bag: Bag, selectorMark: SelectorMark);
+	bag: Bag;
```
Comment (klumba12) [inline src/core/dom/selector/selector.factory.d.ts:7]: private

File: src/core/dom/selector/selector.factory.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+	selectorMark: SelectorMark;
```
Comment (klumba12) [inline src/core/dom/selector/selector.factory.d.ts:8]: private

File: src/core/dom/selector/selector.factory.spec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		'q-grid': 'q-grid',
+		'q-left': 'q-left',
+		'q-right': 'q-right'
+	};
+
+	const selectorMark = new SelectorMark(model, markup, name);
+	const bag = new Bag();
+	const selectorFactory = new SelectorFactory(bag, selectorMark);
+	const foo = (selector, indexA, indexB) => {
+		return {
```
Comment (vkorolev) [inline src/core/dom/selector/selector.factory.spec.js]: can be simplified:
```javascript
{
  selector,
  indexA,
  indexB
}
```

File: src/core/dom/selector/selector.factory.spec.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			const result1 = func(foo);
+			expect(result1.selector.element).to.equal('q');
+			expect(result1.selector.bag.rows.size).to.equal(0);
+			expect(result1.selector.bag.cells.size).to.equal(0);
+			expect(result1.selector.bag.models.size).to.equal(0);
+			expect(result1.selector.factory.columnRange.start).to.equal(1);
+			expect(result1.selector.factory.columnRange.end).to.equal(2);
+			expect(result1.selector.factory.rowRange.start).to.equal(0);
+			expect(result1.selector.factory.rowRange.end).to.equal(3);
+			const f2 = invokes[1];
```
Comment (vkorolev) [inline src/core/dom/selector/selector.factory.spec.js]: consider to move all f2-related stuff to a separate test case and to describe it properly.

**Pull request closed without merging**